src/lava/lib/optimization/problems/problems.py

```diff
@@ -71,16 +72,16 @@ def __init__(self, q: npt.ArrayLike):
         self.validate_input(q)
         self.q = q
         self._q_cost = Cost(q)
-        self._b_variables = DiscreteVariables(domains=[2] * q.shape[0])
+        self._variables._discrete = DiscreteVariables(domains=[2] * q.shape[0])
 
     @property
     def variables(self):
         """Binary variables of the QUBO problem."""
-        return self._b_variables
+        return self._variables
 
     @property
     def num_variables(self):

```

Comment:
from num_variables, I would assume to find the total number of variables, not just the discrete variables. Should be fine to implement it as such, should not break any previous code in the INRC since for the QUBO solver there are only discrete variables.

Reply:
what do you mean? It's a QUBO problem. It has only binary variables by definition